### PR TITLE
Change several deprecated symbols in _macosx.m

### PR DESCRIPTION
**Pull request by jkseppan**
The compiler warns that some of these have been deprecated in
MacOS 10.12, some in 10.14.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jkseppan** [comment]
Right, the new symbols probably don't exist in 10.11.

It looks like Travis can run builds in OSX 10.11 so once the minimum version is decided, perhaps we should test the build in that version: https://docs.travis-ci.com/user/reference/osx/#macos-version

**Comment by jkseppan** [comment]
I think that second commit is the way to stay compatible, but I can't actually properly test myself. I recently had to upgrade my remaining 10.13 system, because various pieces of software were dropping support.

**Comment by dopplershift** [comment]
Unfortunately, due to `@available` I don't think we'll build on anything earlier than the `xcode9` image, which is macOS 10.12. Might be worth giving that a shot here and seeing where we are.

👍 to the changes.

**Comment by jkseppan** [comment]
Converting to draft until I can figure out the CI build

**Comment by tacaswell** [comment]
> Unfortunately, due to @available I don't think we'll build on anything earlier than the xcode9 image, which is macOS 10.12. Might be worth giving that a shot here and seeing where we are.

Given that macports is happy (for now) to patch us and are considering moving to build on newer systems to target older systems this seems like a good place to draw the line.

**Comment by tacaswell** [comment]
Thanks @jkseppan !

**Comment by QuLogic** [comment]
This does not build wheels when targeting 10.9: https://github.com/matplotlib/matplotlib/actions/runs/221342241

**Comment by dopplershift** [comment]
I thought we weren't ourselves supporting building on 10.9?

**Comment by QuLogic** [comment]
Hmm, well the MacPython wheels still specify 10.9, but maybe we determined that that had no effect? And now cibuildwheel is correctly passing that along?

**Comment by tacaswell** [comment]
I'm very confused by this though.  The PR is doing

```
#define CGContext                            graphicsPort
```

but the error we get is:

```
  src/_macosx.m:1661:59: error: 'CGContext' is only available on macOS 10.10 or newer [-Werror,-Wunguarded-availability]
1196
      CGContextRef cr = [[NSGraphicsContext currentContext] CGContext];
1197
                                                            ^~~~~~~~~~
1198
  /Applications/Xcode_11.6.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSGraphicsContext.h:81:35: note: 'CGContext' has been marked as being introduced in macOS 10.10 here, but the deployment target is macOS 10.9.0
1199
  @property (readonly) CGContextRef CGContext NS_RETURNS_INNER_POINTER API_AVAILABLE(macos(10.10));
1200

```

which makes me worry I don't understand what `#define` does in objective-c....

**Comment by tacaswell** [comment]
or at least the order in which the version checking vs marco expansion happens...

File: src/_macosx.m [hunk 1]
```diff
@@ -18,6 +18,40 @@
 #define COMPILING_FOR_10_10
 #endif
 
+#if __MAC_OS_X_VERSION_MAX_ALLOWED < 101200
```
Comment (QuLogic) [inline src/_macosx.m:21]: Should you not be checking for minimum required, like above?

**Comment by QuLogic** [comment]
So what do we want to do about this? Target a newer macOS version? It also breaks building on the shared Mac mini.

**Comment by jkseppan** [comment]
I think we should make a decision about the minimum version of macOS that we support (#18066). Based on this comment by @tacaswell I would suggest making that 10.12:

> > Unfortunately, due to @available I don't think we'll build on anything earlier than the xcode9 image, which is macOS 10.12. Might be worth giving that a shot here and seeing where we are.
> 
> Given that macports is happy (for now) to patch us and are considering moving to build on newer systems to target older systems this seems like a good place to draw the line.

Whichever way that's decided, we can build wheels targeting the supported version and up, and fix the build as needed.

**Pull request merged**